TerriaJS/TerriaMap
# Fix plugin initialization.

Plugins must be initialized before restoring app state from share data as it might reference
plugin components and catalog items.

Depends on https://github.com/TerriaJS/terriajs/pull/6342

### Test me

- Open http://ci.terria.io/main/ 
- Check the map settings, it'll be in `3d smooth`
- Switch to `3d terrain`
- Now reload the page and check map settings again. Correct behavior is the switch to `3d terrain` must persist. But now the settings is incorrectly set to `3d smooth`.

- Now run an instance from this branch `fix-plugin-init`
- The above incorrect behavior should be fixed

- Next, try opening the sample plugin from the right hand map menu (Cube icon)
- Verify it opens correctly without errors
